code/game/turfs/simulated/chasm.dm

```diff
@@ -39,9 +39,9 @@
 		//Portals aren't affected by gravity. Probably.
 		return 0
 	//Flies right over the chasm
-	if(isanimal(AM))
-		var/mob/living/simple_animal/SA = AM
-		if(SA.flying)
+	if(ismob(AM))
```

Comment:
No, line 34 checks for living mobs already (there is potential to re-write the logic in this proc to be more efficient), so anything that gets to this point is either an object or a living mob.